### PR TITLE
Fix infinite loading

### LINKED ISSUE #387: Infinite loading on app launch
**Describe the bug**
Sorry if this issue already exist but I can't find the issue link 😅 There is a handful of users that can't open the app because there is an infinite loading happening on startup.

**To Reproduce**
1. Click on ÉTSMobile app
2. See error

**Expected behavior**
It should redirect to the login page momentarily. The page is probably stuck on the same loading.

**Screenshots**
image

**Smartphone (please complete the following information):**
 - Device: Some android

**Additional context**
I finally found this thread that explain the problem and the solution: https://github.com/mogol/flutter_secure_storage/issues/43

### PR DESCRIPTION
**Pull request by MysticFragilist**
Soooo I think I finally manage to find the bug that was making the cache throw an exception on launch (making the app unusable except if we cleared up the cache and storage of the app and starting over). :smile: @apomalyn look at [this thread](https://github.com/mogol/flutter_secure_storage/issues/43) in flutter_secure_storage repo! It's basically the same error that we have when an error occurs on startup. If this PR don't fix it, we will probably need to review the cache manager entirely. I have hopes tho!

### What this means
Right now, if a new version comes out and this bug happens on a device, the secure storage will be entirely wiped which means the user will have to login again.

However, it should prevent any new app download from having this issue from now on because of the change in the AndroidManifest.xml.

closes #387

### CODE REVIEW

**Review by eneiss — APPROVED** [review]
LGTM! Hope this will fix the issue 😊

**Review by eneiss — APPROVED** [review]
Looks even better to me 😆💯

**Review by apomalyn — CHANGES_REQUESTED** [review]
LGTM!! But... 
*put testing hat*
Could you modify the test already in place for the UserRepository so we could check the storage is correctly deleted when the exception is raised please 😄

**Comment by apomalyn** [comment]
@MysticFragilist I just changed my phone and I'm hitting this issue, so when this will be available I could test it 😉

**Comment by MysticFragilist** [comment]
> @MysticFragilist I just changed my phone and I'm hitting this issue, so when this will be available I could test it 😉 

Oh niiiice!

Je vais faire les tests en fin de semaine, ça devrait pas me prendre beaucoup de temps, je ferai une release tout de suite après :smile:

**Review by AlexandreMarkus — DISMISSED** [review]

File: lib/core/viewmodels/profile_viewmodel.dart [hunk 1]
```diff
@@ -11,7 +11,7 @@ import 'package:notredame/core/managers/user_repository.dart';
 import 'package:ets_api_clients/models.dart';
 
 // OTHERS
-import '../../locator.dart';
+import 'package:notredame/locator.dart';
```
Comment (AlexandreMarkus) [inline lib/core/viewmodels/profile_viewmodel.dart:14]: Good catch

**Review by AlexandreMarkus — COMMENTED** [review]
I have a question, would it be possible, if the error occurs, to delete the secure storage but instead of asking the user to login again, try to login automatically since we already have the username and password in variables?

**Comment by MysticFragilist** [comment]
> I have a question, would it be possible, if the error occurs, to delete the secure storage but instead of asking the user to login again, try to login automatically since we already have the username and password in variables?

Yes it would be a great idea! I'll create the issue for this :) though I won't wait for this feature to be implemented as the bug is somewhat critical (some user can't even login and is probably the main cause for bad reviews right now 😅)

**Review by AlexandreMarkus — APPROVED** [review]

**Review by apomalyn — APPROVED** [review]
LGTM!!!

**Pull request merged**